### PR TITLE
Fix MONGOID-4982 Cannot update a field on embedded association member, add another member and add a nested association to the first member in the same save call

### PR DESCRIPTION
**Pull request by johnnyshields**
Fixes MONGOID-4982

### CODE REVIEW

File: spec/mongoid/persistable/savable_spec.rb [hunk 1]
```diff
@@ -270,31 +270,13 @@
...
-          #         "name" => "Teddy bear"
-          #       }
-          #     },
-          #     "id" => crate.id.to_s
-          #   },
-          #   "1" => {
-          #     "volume" => 0.8
-          #   }
-          # }
-
```
Comment (p-mongo) [inline spec/mongoid/persistable/savable_spec.rb:288]: Why is this removed?

Comment (johnnyshields) [reply spec/mongoid/persistable/savable_spec.rb:288]: It's just a dead-weight comment. Why do we need it?

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/MONGOID-4982

**Review by durran — APPROVED** [review]
One thing we should look into for the future is to clean up all the conflicting updates logic once Mongoid has a minimum mongodb version of 3.6. With the array updates feature we can possible get rid of all this. But for now LGTM.

**Comment by p-mongo** [comment]
We have a follow-up ticket to this in https://jira.mongodb.org/browse/MONGOID-5026, I transferred your comment there. thanks for pointing this out.
